src/utils/resize-observer-wrapper.js

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// This implements a wrapper to the ResizeObserver API, so that only one copy of
+// a ResizeObserver exists. This is much more performant than having one
+// ResizeObserver for each use.
+// This was inspired by the code in https://github.com/jaredLunde/react-hook/blob/master/packages/resize-observer/src/index.tsx
+
+function createResizeObserverWrapper() {
+  const callbacks = new Map();
+  const resizeObserver = new ResizeObserver((entries) => {
+    for (const entry of entries) {
+      const callbacksForElement = callbacks.get(entry.target);
+      if (callbacksForElement) {
+        callbacksForElement.forEach((callback) => callback());
+      }
+    }
+  });
+
+  return {
+    subscribe(element: HTMLElement, callback: () => mixed) {
+      const callbacksForElement = callbacks.get(element) ?? new Set();
+      callbacks.set(element, callbacksForElement);
+      callbacksForElement.add(callback);
+      resizeObserver.observe(element);
+    },
+    unsubscribe(element: HTMLElement, callback: () => mixed) {
+      const callbacksForElement = callbacks.get(element);
+      if (callbacksForElement) {
```

Comment:
nit: This should always be present, considering that we set the subscribers correctly, right? I'm wondering if we should output a warning for the else case of this?